controllers/CtrlNews.php

```diff
@@ -4,47 +4,52 @@
 
 use MVC\Router;
 use Model\NewsModel;
-use SimplePie\SimplePie;
 use Model\FeedModel;
+use Model\CategoriesModel;
+use SimplePie\SimplePie;
+use Controllers\CtrlCategories;
+use Controllers\CtrlNewsCategory;
 
 class CtrlNews{
 
-    public static function getNews(){
-
-    }
-
-    public static function updateNews(Router $router){
-
-    }
 
     public static function registerNews(SimplePie $feed, FeedModel $feeddb ){
         $newsdb = new NewsModel;
         $feedId = $feeddb->id;
         $newsdb->resetAutoIncrement(); 
         foreach ($feed -> get_items() as $item){
-            $newsdb->newsTitle = html_entity_decode($item->get_title());
+            $alerts = [];
+            $exists = NewsModel::where('newsUrl', $item->get_permalink()); 
             
+            if (!$exists) {
+                
+            $newsdb->newsTitle = html_entity_decode($item->get_title());
             $newsdb->newsDescription = html_entity_decode($item->get_description());
-            
             $newsdb->newsDate = $item->get_date('Y-m-d');
-            
             $newsdb->newsUrl = $item->get_permalink();
             
             if ($item->get_enclosure()->get_link() == null) {
                 $newsdb->newsImageUrl = 'no hay imagen disponible';
             }else {
                 $newsdb->newsImageUrl = $item->get_enclosure()->get_link();
-            }   
+            } 
             $newsdb->feedId = $feedId;
             $newsdb->sincroniceEntity();
-            $alerts = $newsdb->validar();
-            if (empty($alerts)) {
-                $newsdb->save();
-                
-        }
+            $alerts = $newsdb->validar();  
+             
+            $newsdb->create();
+            $newdb = NewsModel::where('newsUrl',  $item->get_permalink());
```

Comment:
Aqui puedes usar el array que retorna `create()` para asignar el id en lugar de realizar una consulta.
```suggestion
            $result = $newsdb->create();
            $newdb->id = $result["id"];
```